The minimum setuptools version was recently bumped to 83:
https://github.com/django/django/commit/8362cdc7efc9c881fdba48ce19e89b9194de7c46
Temporarily allow older versions as well.

--- django-5.2.17/pyproject.toml
+++ django-5.2.17/pyproject.toml
@@ -1,5 +1,5 @@
 [build-system]
-requires = ["setuptools>=83"]
+requires = ["setuptools>=80"]
 build-backend = "setuptools.build_meta"
 
 [project]
